test/integration/agent_hosting_hook_deploy_test.go

```diff
@@ -156,6 +156,7 @@ var _ = ginkgo.Describe("Agent hook deploy", func() {
 				Annotations: map[string]string{
 					addonapiv1alpha1.HostingClusterNameAnnotationKey: hostingClusterName,
 				},
+				Finalizers: []string{"pending"}, // this finalizer is to pend addon from deleted for test,will be deleted at the end.
```

Comment:
```suggestion
				Finalizers: []string{"pending"}, // this finalizer is to prevent the addon from being deleted for test, it will be deleted at the end.
```